pkg/operator/operator.go

```diff
@@ -229,16 +228,36 @@ func New(ctx context.Context, cfg *OperatorConfiguration) (*Operator, error) {
 			Metrics:                metricsOpts,
 			HealthProbeBindAddress: cfg.HealthProbeAddr,
 			PprofBindAddress:       "127.0.0.1:8083",
+			Cache: cache.Options{
+				DefaultLabelSelector: labels.SelectorFromSet(map[string]string{ctrlutil.ResourceLabel: ctrlutil.OpName}),
```

Comment:
should we explain the rationale for the default label selector and why there are exceptions for certain resources?

Reply:
Very good point. added.
Additionally I added ClusterObservability and further narrowed down the Service exception.